File: API/GestaoImoveisAPI/Domain/Leasing/ReadjustmentRecord.cs
using GestaoImoveisAPI.Domain.Shared;

namespace GestaoImoveisAPI.Domain.Leasing
{
    public sealed class ReadjustmentRecord : Entity
    {
        public int ContractId { get; private set; }
        public EconomicIndex Index { get; private set; }
        public decimal Rate { get; private set; }
        public decimal OldValue { get; private set; }
        public decimal NewValue { get; private set; }
        public DateTime AppliedAt { get; private set; }

        private ReadjustmentRecord() { } // EF Core

        internal static ReadjustmentRecord Create(
            int contractId,
            EconomicIndex index,
            decimal rate,
            decimal oldValue,
            decimal newValue)
        {
            return new ReadjustmentRecord
            {
                ContractId = contractId,
                Index = index,
                Rate = rate,
                OldValue = oldValue,
                NewValue = newValue,
                AppliedAt = DateTime.UtcNow
            };
        }
    }
}

File: gestao_imoveis_app/lib/features/leasing/domain/entities/readjustment_record.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:gestao_imoveis/features/leasing/domain/value_objects/economic_index.dart';

part 'readjustment_record.freezed.dart';

@freezed
class ReadjustmentRecord with _$ReadjustmentRecord {
  const factory ReadjustmentRecord({
    required int contractId,
    required EconomicIndex index,
    required double rate,
    required double oldValue,
    required double newValue,
    required DateTime appliedAt,
  }) = _ReadjustmentRecord;
}
